src/components/Auth/CP/CPLoginPage.tsx

```diff
@@ -0,0 +1,109 @@
+import styled from '@emotion/styled';
+import CPLoginButton from 'Auth/CP/CPLoginButton';
+import CPLoginStatusText from 'Auth/CP/CPLoginStatusText';
+import CPAuth from 'lib/CPAuth/CPAuth';
+import { FlashMessage, messageTTL, messageType } from 'lib/flashMessage';
+import React, { useCallback, useEffect } from 'react';
+import { Breadcrumb } from 'react-breadcrumbs';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppRoutes } from 'shared/constants/routes';
+import DocumentTitle from 'shared/DocumentTitle';
+import { getCPAuthUser } from 'stores/cpauth/selectors';
+
+const CPStatus = styled.div`
+  margin-top: 32px;
+  max-width: 1024px;
+  display: flex;
+  align-items: center;
+`;
+
+interface ICPLoginPageProps {}
+
+const CPLoginPage: React.FC<ICPLoginPageProps> = () => {
+  const dispatch = useDispatch();
+
+  const handleLogin = useCallback(
+    async (e: React.MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault();
+      e.stopPropagation();
+
+      try {
+        const auth = CPAuth.getInstance();
+        await auth.attemptLogin();
+      } catch (err) {
+        new FlashMessage(err.message, messageType.ERROR, messageTTL.MEDIUM);
+      }
+    },
+    []
+  );
+
+  const handleLogout = useCallback(
+    async (e: React.MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault();
+      e.stopPropagation();
+
+      try {
+        const auth = CPAuth.getInstance();
+        await auth.logout();
+
+        // Force a reload, so we could re-run the batched actions.
+        window.location.href = AppRoutes.Home;
+      } catch (err) {
+        new FlashMessage(err.message, messageType.ERROR, messageTTL.MEDIUM);
+      }
+    },
+    []
+  );
+
+  useEffect(() => {
+    const currentURL = window.location.href;
+
+    if (currentURL.includes(AppRoutes.CPAccessCallback)) {
+      const handleAuthParams = async () => {
+        try {
+          const auth = CPAuth.getInstance();
+          await auth.handleLoginResponse(currentURL);
```

Comment:
This is ran after the user was redirected back to the app from the OAuth2 provider. The login token and state are then taken from the URL parameters.